### PR TITLE
Increase api latency threshold for cluster-scoped list calls

### PR DESCRIPTION
**Pull request by shyamjvs**
Recent change from @smarterclayton (https://github.com/kubernetes/kubernetes/pull/52237) added scope to apiserver metrics. As a result, our current threshold for list calls is no longer sufficient for all-namespace calls which are now being measured separately from namespaced lists. For e.g (from our [last 5k run](https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-scale-performance/37)):

```
WARNING Top latency metric: {Resource:pods Subresource: Verb:LIST Scope:cluster Latency:{Perc50:4.498374s Perc90:7.548079s Perc99:8.169389s Perc100:0s} Count:1400}
```

cc @kubernetes/sig-scalability-misc @kubernetes/sig-api-machinery-misc @wojtek-t

### CODE REVIEW

**Comment by dims** [comment]
@liggitt @smarterclayton - Can one of you PTAL?

File: test/e2e/framework/metrics_util.go [hunk 1]
```diff
@@ -52,9 +52,11 @@ const (
 	apiCallLatencyThreshold time.Duration = 1 * time.Second
 
 	// We use a higher threshold for list apicalls if the cluster is big (i.e having > 500 nodes)
-	// as list response sizes are bigger in general for big clusters.
-	apiListCallLatencyThreshold  time.Duration = 5 * time.Second
-	bigClusterNodeCountThreshold               = 500
+	// as list response sizes are bigger in general for big clusters. We also use a higher threshold
+	// for list calls with cluster scope (all namespaces).
+	apiListCallLatencyThreshold      time.Duration = 5 * time.Second
+	apiClusterScopeListCallThreshold time.Duration = 10 * time.Second
```
Comment (lavalamp) [inline test/e2e/framework/metrics_util.go]: Do you literally mean cluster scoped (i.e., node objects) or cross-namespace lists? If the latter, please choose a less confusing name.

File: test/e2e/framework/metrics_util.go [hunk 2]
```diff
@@ -371,9 +373,13 @@ func HighLatencyRequests(c clientset.Interface, nodeCount int) (int, *APIRespons
...
 		latency := metrics.APICalls[i].Latency.Perc99
 		isListCall := (metrics.APICalls[i].Verb == "LIST")
+		isClusterScopedCall := (metrics.APICalls[i].Scope == "cluster")
 		isBad := false
 		if latency > apiCallLatencyThreshold {
-			if !isListCall || !isBigCluster || (latency > apiListCallLatencyThreshold) {
+			if !isListCall ||
+				!isBigCluster ||
+				(!isClusterScopedCall && latency > apiListCallLatencyThreshold) ||
+				(latency > apiClusterScopeListCallThreshold) {
```
Comment (lavalamp) [inline test/e2e/framework/metrics_util.go]: This is a really confusing logic statement. I suggest one if/switch to set the threshold, and then this if should read `if latency > threshold`.

File: test/e2e/framework/metrics_util.go (hunk 1 above)
Comment (shyamjvs) [reply test/e2e/framework/metrics_util.go]: I meant the former (which includes both non-namespaced and all-namespaced calls) - changed the comment to make it clearer.

File: test/e2e/framework/metrics_util.go (hunk 2 above)
Comment (shyamjvs) [reply test/e2e/framework/metrics_util.go]: Changed it. LG?

**Comment by smarterclayton** [comment]
Yay, more precise metrics!  :)

**Comment by mbohlool** [comment]
cc @jpbetz

**Pull request merged**